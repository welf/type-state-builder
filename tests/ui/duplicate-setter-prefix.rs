use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct DuplicateSetterPrefix {
    // This should be an error: duplicate setter_prefix attributes
    #[builder(setter_prefix = "with_")]
    #[builder(setter_prefix = "set_")]
    name: String,

    email: String,
}

fn main() {}
