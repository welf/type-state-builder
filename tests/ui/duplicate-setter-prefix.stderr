error: Duplicate setter_prefix attribute. Only one setter_prefix is allowed per field
 --> tests/ui/duplicate-setter-prefix.rs:7:15
  |
7 |     #[builder(setter_prefix = "set_")]
  |               ^^^^^^^^^^^^^^^^^^^^^^
